rosetta-lang/src/main/java/com/regnosys/rosetta/formatting2/XtextResourceFormatter.java

```diff
@@ -47,30 +45,39 @@ public void formatCollection(Collection<Resource> resources, ITypedPreferenceVal
 	}
 
 	@Override
-	public void formatXtextResource(XtextResource resource, ITypedPreferenceValues preferenceValues) {
+	public void formatXtextResource(XtextResource resource, ITypedPreferenceValues preferenceValues,
+			BiConsumer<Resource, String> handler) {
+		LOGGER.info("Formatting file at location " + resource.getURI());
+
 		// setup request and formatter
 		FormatterRequest req = formatterRequestProvider.get();
 		req.setPreferences(preferenceValues);
 		IFormatter2 formatter = iFormatter2Provider.get();
 
-		ITextRegionAccess regionAccess = regionBuilder.forNodeModel(resource).create();
+		ITextRegionAccess regionAccess = null;
+		try {
+			regionAccess = regionBuilder.forNodeModel(resource).create();
+		} catch (Exception e) {
+			LOGGER.info("Resource " + resource.getURI() + " is empty.", e);
+			return;
+		}
+
 		req.setTextRegionAccess(regionAccess);
 
 		// list contains all the replacements which should be applied to resource
-		List<ITextReplacement> replacements = formatter.format(req);
+		List<ITextReplacement> replacements = new ArrayList<>();
```

Comment:
Small optimization: put the initializer in the `catch` branch instead. That way you don't create an additional new array list if the formatting succeeds.